utilities/counted_fs.h

```diff
@@ -0,0 +1,131 @@
+//  Copyright (c) 2016-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <atomic>
+#include <memory>
+
+#include "rocksdb/file_system.h"
+#include "rocksdb/io_status.h"
+#include "rocksdb/rocksdb_namespace.h"
+
+namespace ROCKSDB_NAMESPACE {
+struct FileOpCounters {
+  std::atomic<int> ops;
+  std::atomic<uint64_t> bytes;
+
+  FileOpCounters() : ops(0), bytes(0) {}
+
+  void Reset() {
+    ops = 0;
+    bytes = 0;
+  }
+  void RecordOp(const IOStatus& io_s, size_t added_bytes) {
+    if (!io_s.IsNotSupported()) {
+      ops.fetch_add(1, std::memory_order_relaxed);
+    }
+    if (io_s.ok()) {
+      bytes.fetch_add(added_bytes, std::memory_order_relaxed);
+    }
+  }
+};
+
+struct FileCounters {
+  static const char* kName() { return "FileCounters"; }
+
+  std::atomic<int> opens;
+  std::atomic<int> deletes;
+  std::atomic<int> renames;
+  std::atomic<int> flushes;
```

Comment:
Probably OK if this is only ever used in unit tests, but I would expect this to overflow if ever used in production. Likely also could in benchmarking.

Reply:
These were taken from ReportFileOpCounters in db_bench_tool.cc.  If they should be uint64_t, I have no problems changing them.

Reply:
uint64_t is my preference here.